core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -190,6 +193,8 @@ private[kafka] abstract class AbstractServerThread(connectionQuotas: ConnectionQ
    */
   protected def startupComplete() = {
     startupLatch.countDown()
+    // Activate the shutdown latch by resetting it to count 1
+    shutdownLatch = new CountDownLatch(1)
```

Comment:
Say you invoke `KafkaServer.shutdown()` right after `KafkaServer.startup()`, there is a potential race condition (although unlikely) where `SocketServer.shutdown` is called before the latch is initialised. Right?